WireAPI/Sources/WireAPI/Network/NetworkService/NetworkService.swift

```diff
@@ -0,0 +1,148 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+final class NetworkService: NSObject {
+
+    private let baseURL: URL
+    private var urlSession: URLSession?
+    private var webSocketsByTask = [URLSessionWebSocketTask: WebSocket]()
+
+    init(baseURL: URL) {
+        self.baseURL = baseURL
+    }
+
+    deinit {
+        urlSession?.invalidateAndCancel()
+    }
+
+    func configure(with urlSession: URLSession) {
+        self.urlSession = urlSession
+    }
```

Comment:
Yes, because the `NetworkService` is the `URLSession`s delegate which can only be set when initializing the session. This means we must initialize the network service first without the session, then the session (passing in the service as the delegate), then link it back to the service.